AirCasting/Utils/AveragingService.swift

```diff
@@ -122,35 +125,60 @@ final class AveragingService: NSObject, ObservableObject {
        return values.map({ $0.value }).reduce(0.0, +) / Double(values.count)
     }
     
+    private func averagedMeasurementFrom(chunk: ArraySlice<MeasurementEntity>, window: AveragingWindow, measurementCount: Int) -> [MeasurementEntity] {
```

Comment:
maybe we should guard against a scenario where chunk.count is lower than measurementCount passed in?

Reply:
It's out of scope here naturally, but I think we shouldn't mix averaging into the `MeasurementEntity` itself. It should be separated using a new CD entity or even a new model. It's always better to have raw data separated from processed data both on DB level and on code level (fewer conflict points for both + easier maintenance).

Reply:
I don't think it's a viable option right now. 
First thing is that when we finish the session we're sending measurements to the server and the server expects and uses the same model as `MeasurementEntity`. I think we should be consistent.
And second, a more trivial argument is that it would require heavy refactoring. We use `MeasurementEntity` in so many places and we would have to make sure we are able to use both `MeasurementEntity` and `AveragedMeasurement`. I don't see the point of doing that before the MVP release.